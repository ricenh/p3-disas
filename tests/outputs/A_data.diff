--- outputs/A_data.txt	2023-11-01 19:33:26.282564868 -0400
+++ expected/A_data.txt	2023-10-10 11:54:35.000000000 -0400
@@ -1,11 +1,6 @@
-Usage: ../y86 <option(s)> mini-elf-file
- Options are:
-  -h      Display usage
-  -H      Show the Mini-ELF header
-  -a      Show all with brief memory
-  -f      Show all with full memory
-  -s      Show the program headers
-  -m      Show the memory contents (brief)
-  -M      Show the memory contents (full)
-  -d      Disassemble code contents
-  -D      Disassemble data contents
+Disassembly of data contents:
+  0x200:                               | .pos 0x200 data
+  0x200: 88 77 66 55 44 33 22 11       |   .quad 0x1122334455667788
+  0x208: 78 56 34 12 00 00 00 00       |   .quad 0x12345678
+  0x210: 42 00 00 00 00 00 00 00       |   .quad 0x42
+
