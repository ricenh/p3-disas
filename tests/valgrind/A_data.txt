==1335666== Memcheck, a memory error detector
==1335666== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1335666== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==1335666== Command: ../y86 -D inputs/data.o
==1335666== 
../y86: invalid option -- 'D'
Usage: ../y86 <option(s)> mini-elf-file
 Options are:
  -h      Display usage
  -H      Show the Mini-ELF header
  -a      Show all with brief memory
  -f      Show all with full memory
  -s      Show the program headers
  -m      Show the memory contents (brief)
  -M      Show the memory contents (full)
  -d      Disassemble code contents
  -D      Disassemble data contents
==1335666== 
==1335666== HEAP SUMMARY:
==1335666==     in use at exit: 0 bytes in 0 blocks
==1335666==   total heap usage: 1 allocs, 1 frees, 8,192 bytes allocated
==1335666== 
==1335666== All heap blocks were freed -- no leaks are possible
==1335666== 
==1335666== For lists of detected and suppressed errors, rerun with: -s
==1335666== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
