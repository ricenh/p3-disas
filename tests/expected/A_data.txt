Disassembly of data contents:
  0x200:                               | .pos 0x200 data
  0x200: 88 77 66 55 44 33 22 11       |   .quad 0x1122334455667788
  0x208: 78 56 34 12 00 00 00 00       |   .quad 0x12345678
  0x210: 42 00 00 00 00 00 00 00       |   .quad 0x42

